src/main.ts

```diff
@@ -132,6 +142,14 @@ export function generateFile(ctx: Context, fileDesc: FileDescriptorProto): [stri
             ${joinCode(staticMethods, { on: ',\n\n' })}
           };
         `);
+
+        if (options.outputTypeRegistry) {
+          const messageTypeRegistry = imp('messageTypeRegistry@./typeRegistry');
+
+          chunks.push(code`
+            ${messageTypeRegistry}.set('${fullTypeName}', ${def(fullName)});
```

Comment:
Do I understand correctly, that `def` always keeps the original name? And only `imp` can do a rename?

Reply:
Removed the `def`.

Reply:
Yeah, you're right that `imp`s will be renamed, and `def`s won't , but `def` is short for `definition` and so is for declaring the symbols that `imp` should be renamed around.

I.e. if I'm `imp`orting a symbol `Foo`, but I also have a `class Foo` in my file's output, if I wrap that definition of foo as `class def(Foo)` it will tell the auto-import logic that `Foo` is taken and to rename the imported to `Foo1` or what not.

Ah yeah, so for here we're referring to (say) `FooMessage` but we can trust/know that the `export const FooMessage = { ... }` already marked/wrapped its definition/declaration of `FooMessage` in a `def`.

Basically we only need one `def` per symbol per file, ideally at that symbols declaration point.